flake.nix

```
@@ -15,130 +15,156 @@
     };
   };
 
-  outputs = inputs@{ flake-parts, crane, nixpkgs, fenix, rust-manifest, self, ... }: flake-parts.lib.mkFlake { inherit inputs; } {
-    systems = import inputs.systems;
-
-    imports = [
-      inputs.flake-parts.flakeModules.easyOverlay
-    ];
-
-    perSystem = { self', pkgs, lib, system, ... }:
-      let
-        cargoToml = lib.importTOML ./Cargo.toml;
-
-        pname = "typst";
-        version = cargoToml.workspace.package.version;
-
-        rust-toolchain = (fenix.packages.${system}.fromManifestFile rust-manifest).defaultToolchain;
-
-        # Crane-based Nix flake configuration.
-        # Based on https://github.com/ipetkov/crane/blob/master/examples/trunk-workspace/flake.nix
-        craneLib = (crane.mkLib pkgs).overrideToolchain rust-toolchain;
-
-        # Typst files to include in the derivation.
-        # Here we include Rust files, docs and tests.
-        src = lib.fileset.toSource {
-          root = ./.;
-          fileset = lib.fileset.unions [
-            ./Cargo.toml
-            ./Cargo.lock
-            ./rustfmt.toml
-            ./crates
-            ./docs
-            ./tests
-          ];
+  outputs =
+    inputs@{
+      flake-parts,
+      crane,
+      nixpkgs,
+      fenix,
+      rust-manifest,
+      self,
+      ...
+    }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = import inputs.systems;
+
+      imports = [
+        inputs.flake-parts.flakeModules.easyOverlay
+      ];
+
+      perSystem =
+        {
+          self',
+          pkgs,
+          lib,
+          system,
+          ...
+        }:
+        let
+          cargoToml = lib.importTOML ./Cargo.toml;
+
+          pname = "typst";
+          version = cargoToml.workspace.package.version;
+
+          rust-toolchain = (fenix.packages.${system}.fromManifestFile rust-manifest).defaultToolchain;
+
+          # Crane-based Nix flake configuration.
+          # Based on https://github.com/ipetkov/crane/blob/master/examples/trunk-workspace/flake.nix
+          craneLib = (crane.mkLib pkgs).overrideToolchain rust-toolchain;
+
+          # Typst files to include in the derivation.
+          # Here we include Rust files, docs and tests.
+          src = lib.fileset.toSource {
+            root = ./.;
+            fileset = lib.fileset.unions [
+              ./Cargo.toml
+              ./Cargo.lock
+              ./rustfmt.toml
+              ./crates
+              ./docs
+              ./tests
+            ];
+          };
+
+          # Typst derivation's args, used within crane's derivation generation
+          # functions.
+          commonCraneArgs = {
+            inherit src pname version;
+
+            buildInputs = [
+              pkgs.openssl
+            ]
+            ++ (lib.optionals pkgs.stdenv.isDarwin [
+              pkgs.libiconv
+            ]);
```

Based on https://github.com/NixOS/nixpkgs/pull/354097/files#diff-9a6b612d1f795112a2502df31b823fcc2b6e396a68b41d16246e247da532549a, I think we can now remove this entirely.

```suggestion
            ];
```

---

I don't have a mac to test, but making this change isn't throwing any errors at least.